ext/sqlite3/sqlite3.c

```diff
@@ -117,8 +117,8 @@ PHP_METHOD(sqlite3, open)
 	if (strlen(filename) != filename_len) {
 		return;
 	}
```

Comment:
I would say yes, ity can be removed.

Also, why not use a "P" , to be given a zend_string ?  Just a small refactor, zend_string should really be prefered over C strings as much as possible.
The comparaisons would then use zend_string_equals_literal().
I can patch it if you want